usr/libexec/greenboot/greenboot

```diff
@@ -16,45 +15,28 @@ source_configuration_file() {
     fi
 }
 
-source_configuration_file
-function init_disabled_map {
-    for disabled_healthcheck in "${DISABLED_HEALTHCHECKS[@]}"; do
-	DIS_CHECK_STATUS["$disabled_healthcheck"]=1;
-    done
-}
-
-source_configuration_file
-function print_unexecuted_checks {
-    for disabled_healthcheck in "${DISABLED_HEALTHCHECKS[@]}"; do
-	if [[  "${DIS_CHECK_STATUS[$disabled_healthcheck]}" == 1 ]]; then
-           echo "WARNING: $disabled_healthcheck wasn't executed because it's disabled"
-	fi
-    done
-}
-
 source_configuration_file
 function is_disabled {
     healthcheck=$1
     for disabled_healthcheck in "${DISABLED_HEALTHCHECKS[@]}"; do
-        if [ "${healthcheck}" == "${disabled_healthcheck}" ]; then
-	    DIS_CHECK_STATUS["${disabled_healthcheck}"]=0
+      if [ "${healthcheck}" == "${disabled_healthcheck}" ]; then
 	    return 0
 	fi
     done
     return 1
 }
 
-init_disabled_map
-
 script_runner () {
   local scripts_dir=$1; shift
   local mode=$1; shift
   local start_msg=$1; shift
   local required_hc_failed=false
   echo "$start_msg"
   for script in $(find "$scripts_dir" -name '*.sh' | sort); do
-    if is_disabled "$(basename "$script")"; then
-      echo "'$(basename "$script")' was skipped, as specified in config"
+    if [[ "$required_hc_failed" == true && "$mode" == "strict" ]]; then
+      echo "<5>'$(basename "$script")' was skipped due to a previous failure"
+    elif is_disabled "$(basename "$script")"; then
+      echo "<5>'$(basename "$script")' was skipped, as specified in config"
     else
       local rc=0
```

Comment:
can this `rc` variable be renamed to something providing clarity what is it meant for , like is it a return code or some count ?